packages/pubsub/src/subscription.js

```diff
@@ -149,154 +144,28 @@ var PUBSUB_API_TIMEOUT = 90000;
-     *     it does not exist. Default: `false`
-     *
-     * @example
-     * subscription.get(function(err, subscription, apiResponse) {
-     *   // `subscription.metadata` has been populated.
-     * });
-     *
-     * //-
-     * // If the callback is omitted, we'll return a Promise.
-     * //-
-     * subscription.get().then(function(data) {
-     *   var subscription = data[0];
-     *   var apiResponse = data[1];
-     * });
-     */
-    get: true,
-
-    /**
-     * Get the metadata for the subscription.
-     *
-     * @resource [Subscriptions: get API Documentation]{@link https://cloud.google.com/pubsub/docs/reference/rest/v1/projects.subscriptions/get}
-     *
-     * @param {function} callback - The callback function.
-     * @param {?error} callback.err - An error returned while making this
-     *     request.
-     * @param {?object} callback.metadata - Metadata of the subscription from
-     *     the API.
-     * @param {object} callback.apiResponse - Raw API response.
-     *
-     * @example
-     * subscription.getMetadata(function(err, metadata, apiResponse) {});
-     *
-     * //-
-     * // If the callback is omitted, we'll return a Promise.
-     * //-
-     * subscription.getMetadata().then(function(data) {
-     *   var metadata = data[0];
-     *   var apiResponse = data[1];
-     * });
-     */
-    getMetadata: {
-      protoOpts: {
-        service: 'Subscriber',
-        method: 'getSubscription'
-      },
-      reqOpts: {
-        subscription: this.name
-      }
-    }
-  };
+  this.pubsub = pubsub;
+  this.request = pubsub.request.bind(pubsub);
+  this.projectId = pubsub.projectId;
 
-  var config = {
-    parent: pubsub,
-    id: this.name,
-    methods: methods
-  };
+  this.name = Subscription.formatName_(pubsub.projectId, name);
+  this.connection = null;
+  this.encoding = options.encoding || 'utf-8';
+  this.ackDeadline = options.ackDeadline || 600;
+  this.messages_ = new Set();
 
-  if (options.topic) {
-    // Only a subscription with knowledge of its topic can be created.
-    config.createMethod = pubsub.subscribe.bind(pubsub, options.topic);
-    delete options.topic;
-
-    /**
-     * Create a subscription.
-     *
-     * **This is only available if you accessed this object through
-     * {module:pubsub/topic#subscription}.**
-     *
-     * @param {object} config - See {module:pubsub#subscribe}.
-     *
-     * @example
-     * subscription.create(function(err, subscription, apiResponse) {
-     *   if (!err) {
-     *     // The subscription was created successfully.
-     *   }
-     * });
-     *
-     * //-
-     * // If the callback is omitted, we'll return a Promise.
-     * //-
-     * subscription.create().then(function(data) {
-     *   var subscription = data[0];
-     *   var apiResponse = data[1];
-     * });
-     */
-    config.methods.create = true;
-  }
+  this.flowControl = extend({
+    highWaterMark: 16,
+    maxMessages: Infinity
```

Comment:
I can go back to `maxBytes` if you feel strongly about it. 

> 16 is not an appropriate default for maxBytes.

For non-object streams the `highWaterMark` value is usually specified in Bytes, however gRPC gives us back an object stream, to which `highWaterMark` then becomes how many objects do you want to store in memory - hence the 16 which is the default for Node object streams.

If you think we need to specify bytes specifically and gRPC allows us to specify bytes for an object stream, then I'm all for moving in that direction.